src/transformers/generation/logits_process.py

```diff
@@ -266,9 +268,8 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
 
         # Remove tokens with cumulative top_p above the threshold (token with 0 are kept)
         sorted_indices_to_remove = cumulative_probs <= (1 - self.top_p)
-        if self.min_tokens_to_keep > 1:
```

Comment:
@amyeroberts -- this line was preventing the application of `min_tokens_to_keep` when it was `1`. Removing it sorts the problem :)

Reply:
e.g. if we were to set `top_p=0.0`, `.generate()` would crash due to the lack of suitable continuations, despite the default of `min_tokens_to_keep` being `1`. 

After this fix, setting `top_p=0.0` does not crash the code. Refresher: Top P selects all the tokens whose cumulative probability is `>= top_p`. Setting it to `0.0` means it should pick exactly one token.

Reply:
Ah, it's a stupid ambiguity of English issue :D From the title I thought it meant 'when `top_p` is 1`.